src/index.ts

```diff
@@ -8,44 +8,63 @@ import {
   CommentBlock,
 } from "babel-types"
 
+const createCommentBlock = (value: string): Comment => ({
+  type: "CommentBlock",
+  value,
+}) as CommentBlock
+
 const PURE_ANNOTATION = "#__PURE__"
 
-const isPureAnnotated = (comments?: Comment[]): boolean => {
-  if (typeof comments === "undefined") {
+const isPureAnnotated = (nodePath: NodePath<Node>): boolean => {
+  const { leadingComments } = nodePath.node
+
+  if (leadingComments === undefined) {
     return false
   }
-  if (comments.length > 0) {
-    return comments[comments.length - 1].value === PURE_ANNOTATION
+
+  return /[@#]__PURE__/.test(leadingComments[leadingComments.length - 1].value)
+}
+
+const annotateAsPure = (nodePath: NodePath<Node>) => {
+  if (isPureAnnotated(nodePath)) {
+    return;
   }
-  return false
+
+  const node = nodePath.node;
+  node.leadingComments = (node.leadingComments || []).concat(createCommentBlock(PURE_ANNOTATION))
 }
 
-const createComponentBlock = (value: string): Comment => ({
-  type: "CommentBlock",
-  value,
-}) as CommentBlock
+const isTopLevel = (nodePath: NodePath<Node>): boolean => {
+  return nodePath.getStatementParent().parentPath.isProgram()
+}
 
 export default () => ({
   inherits: syntax,
   visitor: {
-    CallExpression: {
-      enter(nodePath: NodePath<Node>) {
-        if (nodePath.parentPath.isVariableDeclarator()
-          || nodePath.parentPath.isAssignmentExpression()
-          || (nodePath.parentPath.isCallExpression()) && ((nodePath.parentPath.get(
-            "arguments") as any) || [] as NodePath[]).indexOf(nodePath) > -1) {
-          if (!isPureAnnotated(nodePath.node.leadingComments)) {
-            const pureAnnotation = createComponentBlock(PURE_ANNOTATION)
-
-            nodePath.replaceWith({
-              ...nodePath.node,
-              leadingComments: nodePath.node.leadingComments
-                ? nodePath.node.leadingComments.concat(pureAnnotation)
-                : [pureAnnotation],
-            })
+    CallExpression(nodePath: NodePath<Node>) {
+      if (!isTopLevel(nodePath)) {
```

Comment:
I still keep option that we can ignore check is a node is top level

Reply:
could u rephrase? I dont understand what u have meant